media-proxy/include/session-rdma.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Intel Corporation
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef __SESSION_RDMA_H
+#define __SESSION_RDMA_H
+
+#include <stdlib.h>
+#include <mcm_dp.h>
+
+#include "libfabric_dev.h"
+#include "libfabric_ep.h"
+#include "session-base.h"
+using namespace std;
+
+typedef struct {
+    memif_buffer_t shm_buf;
+    bool used;
+} shm_buf_info_t;
+
+typedef struct {
+    size_t transfer_size;
+    rdma_addr remote_addr;
+    rdma_addr local_addr;
+    enum direction dir;
+} rdma_s_ops_t;
+
+class tx_rdma_session : public session
+{
+  public:
+    libfabric_ctx *rdma_ctx;
+    ep_ctx_t *ep_ctx;
+
+    volatile bool stop;
```

Comment:
The fact that you use `volatile` is really suspicious. I'm not insisting on reworking that immediately, but I strongly recommend to consider using an atomic variable.